littleX_BE/littleX.jac

```diff
@@ -0,0 +1,230 @@
+        tweet_node = &here.tweet_id;
+        tweet_node <+:like():+> self;
+        logger.info(f"Liked a tweet: {tweet_node}");
+        report tweet_node;
+    }
+
+    can remove_like with remove_like entry {
+        tweet_node = &here.tweet_id;
+        tweet_node del-:like:-> self;
+        logger.info(f"Removed like from tweet: {tweet_node}");
+        report tweet_node;
+    }
+}
+
+node tweet {
+    has content: str;
+    has created_at: str = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S");
+    has tweet_info: dict = {};
+
+    can update with update_tweet exit {
+        self.content = here.updated_content;
+        logger.info(f"Tweet updated: {self}");
+        report self;
+    }
+
+    can delete with remove_tweet exit {
+        logger.info(f"Tweet deleted: {self}");
+        del self;
+    }
+
+    can load with load_tweets entry {
+        logger.info(f"Tweet loaded: {here}");
+        self.tweet_info = {
+            "id": jid(self),
+            "content": self.content,
+            "likes": [i.username for i in [self-:like:->]],
+            "comments": [{"id": jid(i), "content": i.content} for i in [self-->(`?comment)]]
+        };
+    }
+}
+
+node comment {
+    has content: str;
+}
+
+edge follow {}
+
+edge like {}
+
+edge post {}
+
+walker user_init {
+    can user_init with `root entry {
+        visit [-->(`?profile)] else {
+            logger.info("Creating new profile");
+            new_profile = here ++> profile();
+            Jac.unrestrict(new_profile[0], level="CONNECT");
+            visit new_profile;
+        }
+    }
+}
+
+walker update_profile :user_init: {
+    has new_username: str;
+}
+
+walker get_profile  :user_init: {}
+
+walker load_user_profiles {
+    obj __specs__ {
+        static has auth: bool = False;
+    }
+    can load_profiles with `root entry {
+        self.profiles: list = [];
+        
+        for user in NodeAnchor.Collection.find({"name": "profile"}) {
+            user_node = user.architype;
+            self.profiles.append(
+                {"name": user_node.username, "id": jid(user_node)}
+            );
+        }
+        report self.profiles;
+    }
+}
+
+walker follow_request :user_init: {
+    has profile_id: str;
+}
+
+walker un_follow_request :user_init: {
+    has profile_id: str;
+}
+
+walker create_tweet :user_init: {
+    has content: str;
+
+    can tweet with profile entry {
+        tweet_node = here +:post:+> tweet(content=self.content);
+        logger.info(f"Creating new tweet: {self.content}");
+        Jac.unrestrict(tweet_node[0], level="CONNECT");
+        report tweet_node;
+    }
+}
+
+walker update_tweet {
+    has updated_content: str;
+}
+
+walker remove_tweet {}
+
+walker like_tweet :user_init: {
+    has tweet_id: str;
+}
+
+walker remove_like :user_init: {
+    has tweet_id: str;
+}
+
+walker comment_tweet :user_init: {
+    has tweet_id: str;
+    has content: str;
+
+    can add_comment with profile entry {
```

Comment:
We can access the commentor's profile via the root node and directly spawn this walker on the tweet node, without needing a tweet_id.

Reply:
Same for like.